core/pref.h

```diff
@@ -4,6 +4,8 @@
 
 #ifdef __cplusplus
 extern "C" {
+#else
+#include <stdbool.h>
```

Comment:
Note that you can include <stdbool.h> in C++, so you don't have to introduce this else branch. On the other hand, it doesn't hurt either.

See: http://en.cppreference.com/w/cpp/header/cstdbool